cmake/Util.cmake

```
@@ -0,0 +1,58 @@
+# Defines a target that depends on ${FILES} and the files found by globbing
+# when using ${GLOB_PAT} and ${GLOB_DIRS}. The target will only run if any of
+# the files it depends on is changed.
+#
+# REQUIRED  - Abort if COMMAND doesn't exist.
+# GLOB_PAT  - Glob pattern to use. Only used if ${GLOB_DIRS} is specified.
+# GLOB_DIRS - The directories to recursively search for files with extension
+#             ${GLOB_PAT}.
+function(add_smart_target)
```

- maybe we should call this a "glob target". And make the name plural, which is a hint that this creates many targets.
- to avoid target explosion, it might be worth exploring the idea of creating a target per-directory instead of per-file. That might still be fast.

```suggestion
function(add_glob_targets)
```

---

Added CREATE_SINGLE_TOUCH to help with the target explosion.

---

Actually this just creates a single target, and a bajillion commands, but I guess that's just a technicality.